src/components/FormProduct/index.jsx

```diff
@@ -10,25 +15,29 @@ export const FormProduct = () => {
 
   const [product, setProduct] = useState(initialStateProduct);
 
-  const [error, setError] = useState(false);
-
+  const [error, setError] = useState('');
   const handleInputProduct = (e) => {
     setProduct({
       ...product,
       [e.target.name]: e.target.value,
     });
+    setError(
+      e.target.value ? (
+        ''
+      ) : (
+        <span aria-live="polite">All fields are required</span>
+      ),
+    ); //Required field. Immediate validation
   };
-
   const handleSubmitProduct = (e) => {
     e.preventDefault();
-    let { item, nextPurchase, lastPurchasedDate } = product;
+    const { item, nextPurchase, lastPurchasedDate } = product;
 
-    if (item.trim() === '' || nextPurchase === 0) {
-      setError(true);
-      return;
-    }
+    //duplication validation
+    const normalizedItemInput = normalizeItem(item);
```

Comment:
I would suggest sending the item as is to the productExists function, and inside that function do the normalizeItem for the input. 
This will result in cleaner code, since you are only doing normalization in a single place, and calling and importing the normalizeItem function in one single file.

Reply:
Done...